artemis-server/src/test/java/org/apache/activemq/artemis/tests/util/ActiveMQTestBase.java

```diff
@@ -235,29 +222,40 @@ public MBeanServer getMBeanServer() {
    // There is a verification about thread leakages. We only fail a single thread when this happens
    private static Set<Thread> alreadyFailedThread = new HashSet<>();
 
-   private LinkedList<RunnableEx> runAfter;
+   private volatile List<RunnableEx> runAfter;
 
    protected synchronized void runAfter(RunnableEx run) {
       Assert.assertNotNull(run);
       if (runAfter == null) {
-         runAfter = new LinkedList();
+         runAfter = Collections.synchronizedList(new ArrayList<>());
       }
       runAfter.add(run);
    }
 
    @After
    public void runAfter() {
-      if (runAfter != null) {
-         runAfter.forEach((r) -> {
+      List<RunnableEx> theRunafterList = runAfter;
+      if (theRunafterList != null) {
+         // I do this replacement as eventually a runAfter lambda could be adding another runAfter.
+         // this shouldn't happen very often but I saw it happening once.
+         // it's better to be protected against
+         runAfter = null;
```

Comment:
If doing the above suggestion, this could just be moved to straight after the initial local-assignment and done once regardless.

The runAfter() method should perhaps be synchronized, as the runAfter(Runnable) method is and yet still isnt safe despite it due to this nulling and the variable being made volatile.